packages/components/src/internal/components/entities/actions.ts

```diff
@@ -110,119 +111,50 @@ export function getSampleOperationConfirmationData(
     });
 }
 
-function getSelectedParents(
+async function getSelectedParents(
     schemaQuery: SchemaQuery,
     filterArray: Filter.IFilter[],
     isAliquotParent?: boolean
 ): Promise<List<EntityParentType>> {
-    return new Promise((resolve, reject) => {
-        const isSampleParent = isSamplesSchema(schemaQuery);
-        let columns = 'LSID,Name,RowId';
-        if (isSampleParent) {
-            columns += ',SampleSet';
-        }
-        return selectRowsDeprecated({
-            schemaName: schemaQuery.schemaName,
-            queryName: schemaQuery.queryName,
-            viewName: schemaQuery.viewName,
-            columns,
-            filterArray,
-        })
-            .then(response => {
-                if (isSampleParent) {
-                    resolve(resolveSampleParentTypes(response, isAliquotParent));
-                } else {
-                    resolve(resolveEntityParentTypeFromIds(schemaQuery, response, isAliquotParent));
-                }
-            })
-            .catch(reason => {
-                console.error("There was a problem getting the selected parents' data", reason);
-                reject(reason);
-            });
-    });
-}
+    const isSampleParent = isSamplesSchema(schemaQuery);
+    const columns = ['LSID', 'Name', 'RowId'];
+    if (isSampleParent) {
+        columns.push('SampleSet');
+    }
 
-export function getSelectedItemSamples(selectedItemIds: string[]): Promise<number[]> {
-    return new Promise((resolve, reject) => {
-        getSelectedData(
-            SCHEMAS.INVENTORY.ITEMS.schemaName,
-            SCHEMAS.INVENTORY.ITEMS.queryName,
-            selectedItemIds,
-            'RowId, MaterialId',
-            undefined,
-            undefined,
-            undefined
-        )
-            .then(response => {
-                const { data } = response;
-                const sampleIds = [];
-                data.forEach(row => {
-                    sampleIds.push(row.getIn(['MaterialId', 'value']));
-                });
-                resolve(sampleIds);
-            })
-            .catch(reason => {
-                console.error(reason);
-                reject(reason);
-            });
-    });
-}
+    const response = await selectRows({ columns, filterArray, schemaQuery });
```

Comment:
Should this have a try-catch around it or have callers been updated to catch themselves? (Same question for all other conversions to use `await` instead of `.then -> .catch`.)

Reply:
The `selectRows` call is already logging an error. With that, the then/catch paradigm previously being used was superfluous as it was just re-throwing the exception.